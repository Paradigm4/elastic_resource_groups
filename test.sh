#!/bin/bash

#Run with redundancy=0
iquery -anq "remove(test_array_1)" > /dev/null 2>&1
iquery -anq "remove(test_array_2)" > /dev/null 2>&1

iquery -anq "create_with_residency(test_array_1, <val:double> [x=1:40:0:10], false, 'instances=0')" > test.out
iquery -anq "store(build(test_array_1, x), test_array_1)" >> test.out
iquery -anq "create array test_array_2 <val:double> [x=1:40:0:10]" >> test.out
iquery -anq "store(build(test_array_2, x), test_array_2)" >> test.out

#NOTE: tried not to make it depend on equi_join and summarize
#Should say 8 chunks on instance 0
iquery -otsv+ -aq "aggregate(filter(cross_join(list('chunk map') as A, filter(list('arrays'), name='test_array_1') as B), A.uaid=B.uaid), count(*), inst)" >> test.out
iquery -otsv+ -aq "op_count(join(test_array_1, test_array_2))" >> test.out
iquery -otsv+ -aq "op_count(filter(join(test_array_1, test_array_2), test_array_1.val <> test_array_2.val))" >> test.out

diff test.out test.expected
