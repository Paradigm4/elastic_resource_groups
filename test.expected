Query was executed successfully
Query was executed successfully
Query was executed successfully
Query was executed successfully
0	8
0	40
0	0
